assayist/processor/container_go_analyzer.py

```diff
@@ -78,11 +91,19 @@ def run(self):
         for unpack, archive in archives:
             with tempfile.TemporaryDirectory() as subsrc:
                 unpack(archive, subsrc)
-                self._process_source_code(source_location, subsrc)
+                try:
+                    self._process_source_code(source_location, subsrc)
+                except RuntimeError as error:
+                    log.exception(failed_src_exc_msg.format(srcdir))
+                    failures.append(failed_src_msg.format(subsrc, error))
 
         # Now claim all the Go executables.
         self._claim_go_executables()
 
+        if failures:
+            raise AnalysisFailure('GoAnalyzer completed with the following error(s): \n  {}'
+                                  .format("\n  ".join(failures)))
```

Comment:
Just out of curiosity, why the trailing whitespace after `\n` :smiley: (also, f-string? :stuck_out_tongue_closed_eyes:)

Reply:
That way the it displays like this:
```
GoAnalyzer completed with the following error(s):
  some short error
  some loooooooooonger error
```

I find the indentation makes it a little easier to read.